zebra-consensus/src/transaction.rs

```diff
@@ -350,13 +337,16 @@ where
                 .boxed();
 
             // TODO: stop ignoring binding signature errors - #1939
-            // async_checks.push(rsp);
+            // shielded_checks.push(rsp);
         }
 
+        // A set of asynchronous checks which must all succeed.
+        let mut async_checks = stream::select(transparent_checks, shielded_checks);
```

Comment:
`select` is going to get complicated when you have 4 streams, one for each pool.

Consider returning a `futures::stream::FuturesUnordered` instead, which implements `IntoIterator`, `Extend`, and `len`:
https://docs.rs/futures/0.3.15/futures/stream/futures_unordered/struct.FuturesUnordered.html#trait-implementations
https://docs.rs/futures/0.3.15/futures/stream/futures_unordered/struct.FuturesUnordered.html#method.len
https://docs.rs/futures/0.3.15/futures/stream/futures_unordered/struct.FuturesUnordered.html#method.len

Or you could just return a vector or slice of futures. (The difference between `CallAll...`, `FuturesUnordered` and `Vec` only matters when you await the stream's futures.)

But using `FuturesUnordered` would be good for testing each function, and for consistency.

Reply:
Done, updated to use `FuturesUnordered` :+1:

I thought of something else though, maybe it would make sense to compare the performance of using `FuturesUnordered` and using `CallAllUnordered` with `select_all`. I'm curious because I think that with `FuturesUnordered` there's one `Box` allocation for each future, while with `select_all` there's one `Box` per stream (i.e., there will only be one per pool in total). Might be a negligible difference, though. This is very low priority, but would it make sense to open a ticket to investigate if there are any gains to justify the cost of changing the code?

Reply:
The performance will vary based on the size of the futures. So it's something that's worth checking, but only after we've implemented all the transaction validation rules. (And done any batching optimisations.)